src/runtime/server/plugins/errorLogger.ts

```
@@ -6,9 +6,11 @@ export default defineNitroPlugin((nitro: any) => {
 
   if (serverLogLevel === 'error' || serverLogLevel === 'info') {
     nitro.hooks.hook('error', (error: any, { event }: any) => {
-      const url = getRequestURL(event)
-      const fullUrl = url.origin + url.pathname + url.search
-      console.error(`[${event?.node.req.method}] ${fullUrl} - ${error}`)
+      if (event) {
```

silently dismissing the error if the event is missing seems not like a good idea?

---

@fago I think that if `event` is undefined, the error is thrown outside of Nitro and is already logged to console. So it's OK.

@rmuit Can you change the if statement to a guard clause? It's good practice. So, like:
```
nitro.hooks.hook('error', (error: any, { event }: any) => {
      if (!event) {
        return
      }
      const url = getRequestURL(event)
      const fullUrl = url.origin + url.pathname + url.search
      console.error(`[${event?.node.req.method}] ${fullUrl} - ${error}`)
    })
```

---

@TurtlBbx Why would go through the nitro hooks then?

Anyway, if so, this is not obvious, so it should be commented in the code to justify why we can simply return here.

---

@fago The underlying error was:

```
ERROR  [uncaughtException] ENOENT: no such file or directory, open './cf.local.pem'    10:16:57 PM

    at Object.openSync (node:fs:561:18)
    at [.... etc] 
```

It was a Node.js filesystem error (node:fs:561:18). Triggered by `fs.readFileSync('./cf.local.pem')`, inside a Nitro route. Or it was maybe from reading the .env file.

Nitro error hook was triggered, without Nitro's `event`. To "catch" this case, we can simply ignore it, by checking if `event` is falsy/undefined.

We could have a comment before the guard clause, something like:
`// Error is logged to STDOUT/STDERR. Skip logging to prevent duplication.`

@fago Do you have a better comment in mind?

---

@TurtlBbx As I understand we want to log requests that go wrong, this means we can ignore errors that are unrelated to requests, for example "... our case".  --> This is how I'd comment it also.  

not sure how those errors are handled usually but I guess it's out of scope for what we want to achieve here.

---

@fago Those errors are handled out-of-the-box by Nuxt/Nitro, so we don't need to worry about them.

What about simply: `// Ignore errors that are unrelated to requests`. That should be enough.

---

@TurtlBbx agreed sounds good!

---

@rmuit Please add the agreed comment before the return.